BootCampV2/Homeworks/ThirdWeek/Services/UserFileManager.cs

```diff
@@ -0,0 +1,258 @@
+        {
+            loggedInAccount = null;
+
+            Account account = GetAccountByUsername(username);
+
+            if (account == null)
+                return LoginResult.AccountWithUsernameDontExists;
+            else if (!password.Equals(account.Password))
+                return LoginResult.InvalidPassword;
+            else
+                loggedInAccount = account;
+
+            return LoginResult.Success;
+        }
+
+        public RegistrationResult Register(string username, string password, string confirmpassword)
+        {
+            Account account = GetAccountByUsername(username);
+
+            if (account != null)
+                return RegistrationResult.InvalidUsername;
+
+            if (!password.Equals(confirmpassword))
+                return RegistrationResult.NotMatchingPassword;
+
+            account = new Account(username, password);
+
+            AddAccountInUserFile(account);
+
+            return RegistrationResult.Success;
+        }
+
+        public bool ValidateUserFile(string path, out List<Account> account, out Exception exception)
+        {
+            account = new List<Account>();
+            exception = null;
+
+            try
+            {
+                account.AddRange(GetAccountsInUserfile(path));
+
+                List<string> usernames = new List<string>();
+
+                foreach (var item in account)
+                {
+                    usernames.Add(item.Username);
+                }
+
+                if(CheckDistinct(usernames))
+                    throw new DuplicateUserCredentialsException();
+
+                if (string.IsNullOrWhiteSpace(ReadUserFile(Path)) || string.IsNullOrEmpty(ReadUserFile(Path)))
+                {
+                    string[] contentlines = {
+                        "#  # -> Command line\n",
+                        "# .U -> Username line\n",
+                        "# .P -> Password line\n",
+                        "#  - -> Start/End line"
+                        };
+                    
+                    WriteUserfile(contentlines);
+                }
+            }
+            catch (InvalidUserFileFormatException)
+            {
+                exception = new InvalidUserFileFormatException();
+                return false;
+            }
+            catch (DuplicateUserCredentialsException)
+            {
+                exception = new DuplicateUserCredentialsException();
+                return false;
+            }
+            catch (FileNotFoundException)
+            {
+                exception = new UserFileNotFoundException();
+                return false;
+            }
+            catch (Exception)
+            {
+                exception = new Exception();
+                return false;
+            }
+
+            return true;
+        }
+
+        private string ReadUserFile(string path)
+        {
+            var content = string.Empty;
```

Comment:
Instead of concatinating strings with +=, use a `StringBuilder`. It's way faster and the way to go when you have either a variable number of lines or 5+ strings that you might want to concatinate.